tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepository.java

```diff
@@ -0,0 +1,124 @@
+    private static final List<String> EVENT_ATTENDANCE_FLAGS = List.of("$accepted", "$rejected", "$tentativelyaccepted");
+
+    private final MessageIdManager messageIdManager;
+    private final SessionProvider sessionProvider;
+
+    @Inject
+    public StandaloneEventAttendanceRepository(MessageIdManager messageIdManager, SessionProvider sessionProvider) {
+        this.messageIdManager = messageIdManager;
+        this.sessionProvider = sessionProvider;
+    }
+
+    @Override
+    public Publisher<AttendanceStatus> getAttendanceStatus(Username username, MessageId messageId) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
+            .map(this::getAttendanceStatusFromFlags);
+    }
+
+    @Override
+    public Publisher<Void> setAttendanceStatus(Username username, MessageId messageId,
+                                               AttendanceStatus attendanceStatus) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
+            .flatMap(this::removeEventAttendanceFlags)
+            .flatMap(flagsWithoutEventAttendanceFlags -> {
+                Flags flagsToSet = new Flags(flagsWithoutEventAttendanceFlags);
+                getFlagFromAttendanceStatus(attendanceStatus)
+                    .ifPresent(flagsToSet::add);
+                return doSetFlags(messageId, flagsToSet, systemMailboxSession);
+            })
+            .then();
+    }
+
+    private Mono<Flags> removeEventAttendanceFlags(Flags flags) {
+        return Mono.fromSupplier(() -> {
+            Flags flagsWithoutEventAttendanceFlags = new Flags();
+            Arrays.stream(flags.getSystemFlags()).forEach(flagsWithoutEventAttendanceFlags::add);
+            Arrays.stream(flags.getUserFlags())
+                .filter(Predicate.not(EVENT_ATTENDANCE_FLAGS::contains))
+                .forEach(flagsWithoutEventAttendanceFlags::add);
+            return flagsWithoutEventAttendanceFlags;
+        });
+    }
+
+    private Optional<String> getFlagFromAttendanceStatus(AttendanceStatus attendanceStatus) {
+        return switch (attendanceStatus) {
+            case Accepted -> Optional.of("$accepted");
+            case Declined -> Optional.of("$rejected");
+            case Tentative -> Optional.of("$tentativelyaccepted");
+            case Delegated, NeedsAction -> Optional.empty();
+        };
+    }
+
+    private AttendanceStatus getAttendanceStatusFromFlags(Flags flags) {
+        long eventAttendanceFlagsCount = Arrays.stream(flags.getUserFlags())
+            .filter(EVENT_ATTENDANCE_FLAGS::contains)
+            .count();
+
+        if (eventAttendanceFlagsCount > 1) {
+            LOGGER.warn("A message should not have more than one event attendance flag");
+            return AttendanceStatus.NeedsAction;
```

Comment:
So `$accepted`, `$rejected` results in `$needs-action`? Weird!

Reply:
We can not tell which is the first because Flags uses a hashtable internally. The defined behavior in tests should be to return any of them.

Reply:
is there a use case for having several flags ? if not, setting a flag could replace the existing one (if any) so that only the most up to date flag remains ?

Reply:
> setting a flag could replace the existing one (if any) so that only the most up to date flag remains ?

True, this is implemented in the setAttendanceStatus but just in case for some unknown reason ( e.g. concurrency issue) we have found that multiple event attendance flags are associated with the same message, we return any of them and log a warning.